util/string.ts

```diff
@@ -0,0 +1,9 @@
+export type Splitter = (char: string, src: string) => undefined | [string, string]
+
+export const [splitFirst, splitLast] = (["indexOf", "lastIndexOf"] as const).map((m) => {
+  return (char, src) => {
+    const i = src[m](char)
+    if (i === -1) return undefined
+    return [src.slice(0, i), src.slice(i + 1)]
+  }
+}) as [Splitter, Splitter]
```

Comment:
Here's a case where I think it's better WET than DRY. But it can be tightened some with a ternary.
```suggestion
export function splitFirst(char: string, src: string) {
  const i = src.indexOf(char)
  return i === -1 ? undefined : [src.slice(0, i), src.slice(i, 1)]
}

export function splitLast(char: string, src: string) {
  const i = src.lastIndexOf(char)
  return i === -1 ? undefined : [src.slice(0, i), src.slice(i, 1)]
}
```